### PR TITLE
Intake openfe graphml

### LINKED ISSUE #1192: Allow direct intake of a serialized GraphML `LigandNetwork` from OpenFe
Sometimes, e.g for running the PLB, we want to be able to ingest a pre-planned `LigandNetwork` directly rather than planning our own.

### PR DESCRIPTION
**Pull request by hmacdope**
Fixes #1192 

## Developers certificate of origin
- [X] I certify that this contribution is covered by the MIT license as defined in our [LICENSE](https://github.com/choderalab/asapdiscovery/blob/main/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

### CODE REVIEW

**Comment by hmacdope** [comment]
@jthorton you may find this useful for running PLB as discussed cc @JenkeScheen

**Comment by hmacdope** [comment]
@jthorton this is ready to go now, would you be able to review?

File: asapdiscovery-alchemy/asapdiscovery/alchemy/schema/fec.py [hunk 1]
```diff
@@ -409,25 +410,30 @@ def create_fec_dataset(
...
+                raise ValueError(
+                    f"{missing} of {len(ligands)} ligands do not have names; names are required for ligands for downstream results handling"
+                )
+        elif not graphml:
+            raise ValueError("No ligands or graphml provided to create the network.")
 
         # start by trying to plan the network
         planned_network = self.network_planner.generate_network(
-            ligands=ligands, central_ligand=central_ligand
+            ligands=ligands, central_ligand=central_ligand, graphml=graphml
```
Comment (jthorton) [inline asapdiscovery-alchemy/asapdiscovery/alchemy/schema/fec.py:436]: Do we still need to go through the network planner here? we could probably avoid adding the extra logic to `generate_network` this way and have a `from_graphml` method on the `PlannedNetwork` class which could have all of the graphml specific checks.

File: asapdiscovery-alchemy/asapdiscovery/alchemy/schema/fec.py [hunk 2]
```diff
@@ -409,25 +410,30 @@ def create_fec_dataset(
...
         # check that all ligands are unique in the series
-        if len(set(ligands)) != len(ligands):
-            count = Counter(ligands)
-            duplicated = [
-                key.compound_name for key, value in count.items() if value > 1
-            ]
-            raise ValueError(
-                f"ligand series contains {len(duplicated)} duplicate ligands: {duplicated}"
-            )
+        if ligands:
```
Comment (jthorton) [inline asapdiscovery-alchemy/asapdiscovery/alchemy/schema/fec.py:413]: We should unpack the ligands from the graphml input here as well and make sure the names are unique

File: asapdiscovery-data/asapdiscovery/data/backend/rdkit.py [hunk 3]
```diff
@@ -147,6 +147,29 @@ def sdf_str_to_rdkit_mol(sdf: str) -> Chem.Mol:
     return ref
 
 
+def rdkit_mol_to_sdf_str(mol: Chem.Mol) -> str:
```
Comment (jthorton) [inline asapdiscovery-data/asapdiscovery/data/backend/rdkit.py:150]: Great idea this is so much easier than picking the rdkit molecule apart!

File: asapdiscovery-alchemy/asapdiscovery/alchemy/schema/fec.py (hunk 2 above)
Comment (hmacdope) [reply asapdiscovery-alchemy/asapdiscovery/alchemy/schema/fec.py:414]: Moved this to the new `from_graphml` constructor.

File: asapdiscovery-alchemy/asapdiscovery/alchemy/tests/test_cli.py [hunk 4]
```diff
@@ -107,6 +107,24 @@ def test_alchemy_plan_missing():
...
+                "plan",
+                "-g",
+                p38_graphml,
+                "-r",
+                p38_protein,
+                "-n",
+                "tst",
+            ],
+        )
+        assert result.exit_code == 0
```
[the remark this answers is not in the document]
Comment (hmacdope) [reply asapdiscovery-alchemy/asapdiscovery/alchemy/tests/test_cli.py:174]: Doneskies.

**Comment by hmacdope** [comment]
@jthorton I think this is ready for another round of review!

**Review by jthorton — APPROVED** [review]
Great job @hmacdope!

**Pull request merged**